master/internal/api_workspace.go

```diff
@@ -519,12 +558,31 @@ func (a *apiServer) deleteWorkspaceNamespaceBinding(ctx context.Context,
 			if err != nil {
 				return err
 			}
+			deletedClusters := []string{}
 			for _, v := range deletedBindings {
+				deletedClusters = append(deletedClusters, v.ClusterName)
 				err = a.m.rm.RemoveEmptyNamespace(v.Namespace, v.ClusterName)
 				if err != nil {
 					return err
 				}
 			}
+			if len(deletedClusters) < len(clusterNames) {
+				switch {
+				case len(clusterNames) == 1 && (clusterNames[0] != ""):
+					return fmt.Errorf("tried to delete default binding for cluster %v", clusterNames[0])
+				case len(clusterNames) == 1:
+					return fmt.Errorf("tried to delete default binding for cluster")
+				default:
+					remainingClusters := ""
+					for _, i := range clusterNames {
+						if !slices.Contains(deletedClusters, i) {
+							remainingClusters = remainingClusters + i + ", "
+						}
+					}
+					return fmt.Errorf("tried to delete default bindings for the following clusters: %v",
+						remainingClusters[:len(remainingClusters)-2])
```

Comment:
Shoudl this be 
`remainingClusters[:len(remainingClusters)-1])` (because the only char we don't want is the comma)?

Reply:
i was accounting for the comma and the space. It prints as "failed to delete namespace binding: tried to delete default bindings for the following clusters: default, extra," if I do -1

Reply:
Gotchyaa! Missed the space 🙈😂